### PR TITLE
fix: replace Inngest app/generation.requested with an async worker function

### PR DESCRIPTION
**Pull request by codeincontext**
## Description

The quiz designer uses inngest functions. We're transitioning away from inngest and it isn't set up as a Vercel integraiton on the new Vercel project

- Extract generation inngest function to a file
- Wrap and call generation code
- Use async background pattern

## How to test

- Go to the quiz designer
- Generations should work after ~10 seconds instead of hanging

### CODE REVIEW

File: packages/core/src/workers/generations/requestGeneration.ts [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+  } catch (e) {
+    await onFailure({
+      error: e as Error,
+      event: { data: { event: { data } } },
+      logger: baseLogger,
+    });
+  }
+}
+
+async function invoke({ data, user }: InvokeArgs) {
```
Comment (codeincontext) [inline packages/core/src/workers/generations/requestGeneration.ts:55]: This function itself is unchanged

File: packages/core/src/workers/generations/requestGeneration.ts [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+  error: Error;
+  event: {
+    data: {
+      event: { data: z.infer<(typeof requestGenerationSchema)["data"]> };
+    };
+  };
+  logger: Logger;
+};
+
+async function onFailure({ error, event, logger }: OnFailureArgs) {
```
Comment (codeincontext) [inline packages/core/src/workers/generations/requestGeneration.ts:395]: This function is unchanged

File: packages/core/src/workers/generations/requestGeneration.ts [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+  try {
+    return await invoke({ data, user });
+  } catch (e) {
+    await onFailure({
+      error: e as Error,
+      event: { data: { event: { data } } },
+      logger: baseLogger,
+    });
+  }
+}
```
Comment (codeincontext) [inline packages/core/src/workers/generations/requestGeneration.ts:53]: This is new

File: packages/core/src/workers/generations/requestGeneration.ts [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+   * ```
+   *
+   * See `waitUntil` documentation in
+   * [Cloudflare](https://developers.cloudflare.com/workers/runtime-apis/handlers/fetch/#contextwaituntil)
+   * and [Vercel](https://vercel.com/docs/functions/edge-middleware/middleware-api#waituntil)
+   * for more details.
+   * ```
+   */
+  pending: Promise<void>;
+};
```
Comment (codeincontext) [inline packages/core/src/workers/generations/requestGeneration.ts:60]: This pattern is inspired by @upstash/ratelimit

File: packages/core/src/workers/generations/requestGeneration.ts [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+        ModerationType.NOT_ENGLISH,
+        {
+          moderationMeta: { englishScores },
+        },
+      );
+      // Return early and don't process generation further
+      return;
+    }
+
+    //logger.info("Running step: Detect profanity");
```
Comment (tomwisecodes) [inline packages/core/src/workers/generations/requestGeneration.ts:212]: Think this was commented out but shall we delete this while we are at it?

**Review by tomwisecodes — APPROVED** [review]
One small comment looks like the correct way to manage this

File: packages/core/src/workers/generations/requestGeneration.ts (hunk 5 above)
Comment (codeincontext) [reply packages/core/src/workers/generations/requestGeneration.ts]: Done

**Pull request merged**